### PR TITLE
adding changes to refactor transaction api

### PR DESCRIPTION
**Pull request by Manik-Jain**
# Description

Changes made to refactor Transaction API to use generic mapper

## Related Issue

https://github.com/eea-oasis/baseline/issues/552

## Motivation and Context

In order to have code sync across modules, we need to use generic mapper for all APIs

## How Has This Been Tested

tested locally, and via test cases

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Comment by skosito** [comment]
@Manik-Jain what about automapper, is there a decision? imo we should decide on that before adding additional code on old mapper

File: examples/bri-3/src/bri/transactions/agents/transactions.agent.ts [hunk 1]
```diff
@@ -58,7 +58,17 @@ export class TransactionAgent {
       throw new BadRequestException(UPDATE_WRONG_STATUS_ERR_MESSAGE);
     }
 
-    return transactionToUpdate;
+    return new Transaction(
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/transactions/agents/transactions.agent.ts:61]: why is this needed when there is already a mapping from prisma model to domain object in the storage agent?

File: examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts [hunk 2]
```diff
@@ -1,30 +1,43 @@
...
 import { Transaction } from '../models/transaction';
+import { getType } from 'tst-reflect';
 
 @Injectable()
 export class TransactionStorageAgent extends PrismaService {
+  constructor(private readonly mapper: Mapper) {
+    super();
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts:14]: Just a note, we deliberately left our the get all endpoint because we are unsure of the scope for this query. I am fine with including it here.

File: examples/bri-3/src/bri/transactions/capabilities/updateTransaction/updateTransactionCommand.handler.ts [hunk 3]
```diff
@@ -10,6 +12,7 @@ export class UpdateTransactionCommandHandler
   constructor(
     private agent: TransactionAgent,
     private storageAgent: TransactionStorageAgent,
+    private readonly mapper: Mapper,
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/transactions/capabilities/updateTransaction/updateTransactionCommand.handler.ts:15]: mapper is not used anywhere

File: examples/bri-3/src/bri/transactions/agents/transactions.agent.ts (hunk 1 above)
Comment (Manik-Jain) [reply examples/bri-3/src/bri/transactions/agents/transactions.agent.ts:61]: this is needed till the time https://github.com/eea-oasis/baseline/issues/553 is resolved, or we have auto-mapper in place, whatever comes first. This will be changed once we have that in place

File: examples/bri-3/src/bri/transactions/capabilities/updateTransaction/updateTransactionCommand.handler.ts (hunk 3 above)
Comment (Manik-Jain) [reply examples/bri-3/src/bri/transactions/capabilities/updateTransaction/updateTransactionCommand.handler.ts]: change done

File: examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts (hunk 2 above)
Comment (Manik-Jain) [reply examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts:14]: > @Manik-Jain what about automapper, is there a decision? imo we should decide on that before adding additional code to current mapper

Yes, we have agreed to change response interfaces to classes and then we can have auto-mapper changes. 
https://github.com/eea-oasis/baseline/issues/558

**Comment by Manik-Jain** [comment]
> @Manik-Jain what about automapper, is there a decision? imo we should decide on that before adding additional code to current mapper

Yes, we have agreed to change response interfaces to classes and then we can have auto-mapper changes.
https://github.com/eea-oasis/baseline/issues/558

**Review by ognjenkurtic — APPROVED** [review]

**Comment by Therecanbeonlyone1969** [comment]
@Manik-Jain @skosito @ognjenkurtic should we then wait until we have the automapper properly implemented before we do this PR?

cc @Kasshern @Ybittan

**Comment by ognjenkurtic** [comment]
> @Manik-Jain @skosito @ognjenkurtic should we then wait until we have the automapper properly implemented before we do this PR?
> 
> cc @Kasshern @Ybittan

@Manik-Jain is best to provide the answer here. If we can do automapper quickly i guess we do not need to work on the old mapper.

**Comment by Manik-Jain** [comment]
Change taken care in automapper PR
